test/jasmine/tests/plot_api_test.js

```diff
@@ -188,65 +181,60 @@ describe('Test graph_obj', function () {
             expect(gd).toEqual(expected);
         });
 
-        it('work when newIndices is undefined', function () {
-            var expected = {
-                data: [
-                    {'name': 'a'},
-                    {'name': 'b'},
-                    {'name': 'c'},
-                    {'name': 'd'}
-                ]
-            };
+        it('should work when newIndices is undefined', function () {
+            var expectedData = [
+                {'name': 'a'},
+                {'name': 'b'},
+                {'name': 'c'},
+                {'name': 'd'}
+            ];
 
             Plotly.addTraces(gd, [{'name': 'c'}, {'name': 'd'}]);
-            expect(gd).toEqual(expected);
+            expect(gd.data).toEqual(expectedData);
```

Comment:
FYI @bpostlethwaite no need to check the whole `gd`, just checking its data is more than enough here.